server/src/main/java/org/elasticsearch/TransportVersion.java

```diff
@@ -57,7 +73,55 @@
  * different version value. If you need to know whether the cluster as a whole speaks a new enough {@link TransportVersion} to understand a
  * newly-added feature, use {@link org.elasticsearch.cluster.ClusterState#getMinTransportVersion}.
  */
-public record TransportVersion(int id) implements VersionId<TransportVersion> {
+public record TransportVersion(String name, int id, TransportVersion nextPatchVersion) implements VersionId<TransportVersion> {
```

Comment:
for followup, we should validate this name is lowercase alphanumeric+underscores